qa/os/src/test/java/org/elasticsearch/packaging/test/PackagingTestCase.java

```diff
@@ -110,6 +116,10 @@ public void setup() throws Exception {
         assumeFalse(failed); // skip rest of tests once one fails
 
         sh = newShell();
+
+        if (installation != null && Files.exists(installation.logs.resolve("elasticsearch.log"))) {
```

Comment:
While I understand the desire to have a minimal log file when debugging failures, unfortunately I think this will hurt more than it  helps. We have historically had failures in packaging tests where we do not fail the test correctly, so several more test cases may be run before an unrelated failure occurs, yet we would then only see the latest log file. I'm open, though, to any mitigation ideas for the above problem, as I do like this change in principle.

Reply:
Hmm. I made this change because I ran into an almost opposite problem, which was that one test neglected to shut Elasticsearch down and the next test failed silently because it could both connect to Elasticsearch (still running from the previous test) and find a successful startup message in the logs (because the logs were not cleared out.)

I will try to think of a clean way to avoid both problems.

Reply:
I think I've got it, at least for Linux. The `comm` utility can take two files and display the lines in File A that are not in File B. So if I make a temporary copy of the log file, I can get only the lines that have been added to the log file since the test started. I just need to make sure it's resilient against log rotation… and figure out how to do it on Windows.

Reply:
Perhaps an alternative would be to rename the log file at the end of each test?

Reply:
Yes, I'm leaning towards either that or accumulating all of the logs in another file. I've also got a java-only implementation of the tactic that I described above, so that we won't have to maintain both Linux and Powershell command-line-fu.

Reply:
I should have done the renaming approach from the start, and that's what I've ended up committing.